core/src/mapping.rs

```diff
@@ -41,6 +46,25 @@ impl<K: OdraType + Hash, V: OdraType + Default> Mapping<K, V> {
     }
 }
 
+impl<K: OdraType + Hash, V: OdraType + Instance> Mapping<K, V> {
+    /// Reads `key` from the storage or the default value is returned.
+    pub fn get_instance(&self, key: &K) -> V {
+        #[cfg(feature = "mock-vm")]
+        {
+            let key_hash = hex::encode(key.ser().unwrap());
+            let namespace = format!("{}_{}", key_hash, self.name);
+            V::instance(&namespace)
+        }
+
+        #[cfg(feature = "casper")]
```

Comment:
There is no need to hex namespace here. Hashing should only happen right after reading from the storage.

Reply:
There is no namespace hashing here. We need to build the path but the `key`  is not a string, even is not strigifiable, so the only thing that came to my mind to have string representation of the key, was to serialize the key and use it's hex representation.